save-preprocessor/src/main/kotlin/org/cqfn/save/preprocessor/controllers/DownloadProjectController.kt

```diff
@@ -96,9 +96,10 @@ class DownloadProjectController(private val configProperties: ConfigProperties)
     @Suppress(
         "TooGenericExceptionCaught",
         "TOO_LONG_FUNCTION",
-        "TOO_MANY_LINES_IN_LAMBDA")
+        "TOO_MANY_LINES_IN_LAMBDA",
+        "UnsafeCallOnNullableType")
     private fun downLoadRepository(executionRequest: ExecutionRequest) {
-        val gitRepository = executionRequest.gitRepository
+        val gitRepository = executionRequest.gitDto!!
```

Comment:
We should validate request in `@RequestMapping` method and return 4xx, if git is null here

Reply:
I think validation project's git url should be in backend, so I added handle for this case in backend, that means, there always will be not-null `gitDto`

Reply:
Because we just pass `ExecutionRequest` from bakend, but in backend we verify that there isn't null git

Reply:
ok, but it doesn't explain, why we can't make this field not nullable

Reply:
Agree, fixed